vmm/src/lib.rs

```diff
@@ -1451,6 +1451,8 @@ impl RequestHandler for Vmm {
 
         self.vm = Some(vm);
 
+        event!("vm", "rebooted");
```

Comment:
Hhm, I don't know what, if any events `vm.boot()?` triggers (or might trigger in the future) - maybe this should be before `vm.boot()?`. I totally get that if that function generated an error the event might look weird - but a reboot was triggered - it just wasn't successful.

Reply:
`vm.boot()` emits an event (vm booting), so I placed this new event at a later point.

I'm fine with placing this at an earlier place, too.

Reply:
@alexellis Do you have an opinion?

Reply:
How about adding a rebooting event at the begin of vm_boot() and a rebooted event in the end, as we did in the boot()?

Reply:
That seems like a reasonable solution.

Reply:
> > How about adding a rebooting event at the begin of vm_boot() and a rebooted event in the end, as we did in the boot()?
> 
> I don't follow here. `vm_boot()` is also used for freshly booting a VM. Putting a rebooting event there looks wrong to me.
> 

You are right. To address this issue, I propose to move the vm "booting"/"booted" event to `Vmm::vm_boot()` (e.g. the caller of  `Vm::boot()`). Now the event for a reboot will be like: "rebooting" -> "shutdown" -> "rebooted".

> There seems to be a consensus on where to put what, but I cannot decode Bo's comment about restore/boot/resize -- there are no events emitted from `vm_restore` or `vm_resize`.
> 

I can see the confusion. Similar to what's done for vm.boot request, the events for resize and restore are reported from `Vm::resize()` and `Vm::restore()` while I would say it makes sense to move these events to `Vmm::vm_resize()` and `Vmm::vm_restore()`.

> Anyway, feel free to force push to this branch if you already know what to do. It is just a few lines of code. That's probably faster than going another round trip. :-)

PR updated. Please take a look. @liuw @rbradford @up2wing

Reply:
LGTM. Thanks.

Reply:
Updated the PR a bit more, e.g. moved vm "shutdown" event. So now we have the following event sequence.

With VM boot: `"vm", "booting"` -> `"vm", "booted"` 
With  `sudo poweroff` on the guest: `"vm", "shutdown"` -> `"vm", "deleted"` -> `"vmm", "shutdown"`
With `sudo reboot` on the guest: `"vm", "rebooting"` -> `"vm", "rebooted"`

@alexellis  Does this fit your need? It would be better if you can give this PR a quick try with your workflow.